services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaController.kt

```diff
@@ -411,6 +424,14 @@ of the token and link will be reset.
         return HankeError.HAI4004
     }
 
+    @ExceptionHandler
+    @ResponseStatus(HttpStatus.NOT_FOUND)
```

Comment:
I don't think this should be a 404. Not found means that the URL for the request had something wrong with it.

Here, we expect every authorized user to have a verified name, so if it's missing it's a problem with Profiili. Maybe this should be a 500 Internal server error?

Reply:
You're right. I changed it to 500.